content/class/Post.md

```diff
@@ -0,0 +1,316 @@
+
+```c#
+public GLib.DateTime date { get; }
+```
+
+The date this Post was posted on the platform.
+
+| API   | Twitter 1    | Twitter 2       | Mastodon        |
+| ----- | ------------ | --------------- | --------------- |
+| Field | `created_at` | `created_at`    | `created_at`    |
+| Type  | string       | ISO 8601 string | ISO 8601 string |
+
+### `url` {#property_url}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public string url { get; }
+```
+
+A url to open the Post on the website of the used provider.
+
+> Twitter urls have to be generated internally, as there is no field for it.
+>
+> So, not really neccessary for Twitter (as it is only `twitter.com`, which can be generated on the fly), but as Mastodon has multiple servers, we need to specify it for it, should it be implemented.
+
+| API   | Mastodon        |
+| ----- | --------------- |
+| Field | `uri` or `url`  |
+| Type  | string          |
+
+### `language` {#property_language}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+The language this post was written in, used for offering an translation.
+
+| API   | Twitter 1     | Twitter 2       | Mastodon       |
+| ----- | ------------- | --------------- | -------------- |
+| Field | `lang`        | `lang`          | `language`     |
+| Type  | BCP-47 string | BCP-47 string   | ISO 639 string |
+
+### `referenced_post` {#property_referenced_tweet}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public Post? referenced_post { get; }
+```
+
+If an post is an repost or quote, this stores the post reposted or quoted.
+
+| API   | Twitter 1                             | Twitter 2           | Mastodon |
+| ----- | ------------------------------------- | ------------------- | -------- |
+| Field | `retweeted_status` or `quoted_status` | `referenced_tweets` | `reblog` |
+| Type  | object                                | array of objects    | object   |
+
+### `thread` {#property_thread}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public Thread? thread { get; }
```

Comment:
We may have to be careful to avoid reference loops between threads and posts (e.g. Vala `weak` references)